entrypoint.sh

```diff
@@ -17,7 +17,7 @@ if [[ -f "build.gradle" ]]; then
   exit 1
 fi
 
-if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]]; then
+if [[ -z "${SCAN_PULL_REQUEST}" || "${GITHUB_EVENT_NAME}" == "pull_request" ]]; then
   EVENT_ACTION=$(jq -r ".action" "${GITHUB_EVENT_PATH}")
   if [[ "${EVENT_ACTION}" != "opened" ]]; then
 	  echo "No need to run analysis. It is already triggered by the push event."
```

Comment:
@sawy @simonbrandhof Why is this block needed? 
You can let the users decide to run the analysis whenever they want by using the `if` conditional.

> You can use the `if` conditional to prevent a step from running unless a condition is met. You can use any supported context and expression to create a conditional.

[See this for more details](https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsif)

Reply:
> @sawy @simonbrandhof Why is this block needed?

I don't know that. The scripts seems to assume that there is a workflow in place that runs on push. We have workflows that only listen to pull_request events and therefore never run the analysis. Hence the new flag to overwrite the behavior.

Reply:
That is a safeguard to prevent analysis of branch when a pull request exists. To be honest that was put in place by fear of having users misunderstanding how branches and pull requests work in SonarCloud. Indeed a branch can be analysed in isolation of its associated pull request.

Removing this condition, and the suggested `SCAN_PULL_REQUEST` variable, is an option that could make things simpler. It would be up to users to correctly configure their workflow.

Does that make sense?

Reply:
@simonbrandhof could you elaborate a bit more? I think I'm part of the misunderstanding users :worried:, sorry. 
If we open a PR, the checkout action will AFAIK ([per default](https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit)) checkout a merge commit that may differ from the branch HEAD. In such case why analysis should be bypassed?

Reply:
@loicalbertin it doesn't actually make much sense at all, but it seems @simonbrandhof doesn't want to change anything either. If you want to have your pull requests scanned properly, try `pr-fix` branch on [my fork](https://github.com/hakimio/sonarcloud-github-action/tree/pr-fix). We've been using it for quite some time now and haven't noticed any issues neither on SonarCloud side, nor on GitHub itself. Everything's working fine.

You can include it in your github workflow the following way:
```yml
      - name: SonarCloud Scan
        uses: hakimio/sonarcloud-github-action@pr-fix
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```